### PR TITLE
Edit Patient Screen Overhaul 1

### PR DESCRIPTION
**Pull request by Cinericius**
*Moved Patient Info Button
*Implemented saving patient info
*Implemented populating patient info fields

Things that may still need to be addressed in the future:
*Renaming patients and things that would be affected by renaming patients, i.e. getting Patient Names for the splash screen

### CODE REVIEW

File: app/src/main/java/com/ad_revenue/damp/Services/JSONService.java [hunk 1]
```diff
@@ -87,6 +87,27 @@ private String getPatientInfo(Context context, String patientName) {
         return toReturn.toArray(new String[toReturn.size()]);
     }
 
+    public ArrayList<String> getPatientInformation(Context context, String patientName) {
+        try {
+            FileReader readMe = new FileReader(getPatientInfo(context, patientName));
+            JSONArray currentJson = (JSONArray) parser.parse(readMe);
```
Comment (Cinericius) [inline app/src/main/java/com/ad_revenue/damp/Services/JSONService.java]: Still not entirely sure this needs to be an array.

File: app/src/main/java/com/ad_revenue/damp/Edit_Patient_Screen.java [hunk 2]
```diff
@@ -7,20 +7,36 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         jsonService = new JSONService();
         context = getApplicationContext();
         setContentView(R.layout.activity_edit__patient__screen);
-
-
+        if (!jsonService.isPatientInfoPresent(context, getIntent().getStringExtra("patientName"))) {
+            properties = new ArrayList<String>(){{
```
Comment (Cinericius) [inline app/src/main/java/com/ad_revenue/damp/Edit_Patient_Screen.java]: Behold the slightly unnecessary anonymous inner class.

File: app/src/main/java/com/ad_revenue/damp/Services/JSONService.java [hunk 3]
```diff
@@ -87,6 +87,27 @@ private String getPatientInfo(Context context, String patientName) {
...
+            readMe.close();
+            return patientInfo;
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.out.println("Error reading patient information.");
+        } catch (ParseException e) {
+            e.printStackTrace();
+            System.out.println("Error parsing patient information.");
+        }
+        return new ArrayList<String>();
```
Comment (Cinericius) [inline app/src/main/java/com/ad_revenue/damp/Services/JSONService.java]: Should this instead return the default properties, or is it fine?

Comment (arodriguezfebres) [reply app/src/main/java/com/ad_revenue/damp/Services/JSONService.java]: If they get to this point, we're going to eventually have to re-route them back to where they came from so for now, this'll do. (BUT, we will be creating failsafes for almost all our catch statements, hopefully during Sprint's 4 and 5.)

File: app/src/main/java/com/ad_revenue/damp/Edit_Patient_Screen.java (hunk 2 above)
Comment (arodriguezfebres) [reply app/src/main/java/com/ad_revenue/damp/Edit_Patient_Screen.java]: This is so unnecessary, yet so necessary. I don't know how to feel. ❓ ❔

File: app/src/main/java/com/ad_revenue/damp/Services/JSONService.java (hunk 1 above)
Comment (arodriguezfebres) [reply app/src/main/java/com/ad_revenue/damp/Services/JSONService.java]: Well, if we fix it for this method, we'll have to rewrite how our reading/writing work as JSONArrays and JSONObjects don't work the same, HOWEVER, considering how many times we repeat this code, I don't know why I haven't made general purpose methods that does this. 

I am taking note of it, hopefully we can clean this kind of stuff up without needing to re-do the Reading/Writing methods entirely.

**Comment by Cinericius** [comment]
For some reason it made me resolve the same conflict twice. That was dumb.

**Pull request merged**